src/modules/commander/PreflightCheck.cpp

```diff
@@ -592,55 +592,59 @@ static bool ekf2Check(orb_advert_t *mavlink_log_pub, vehicle_status_s &vehicle_s
 	if (fabsf(status.states[10]) > test_limit || fabsf(status.states[11]) > test_limit
 	    || fabsf(status.states[12]) > test_limit) {
 		if (report_fail) {
-			mavlink_log_critical(mavlink_log_pub, "PREFLIGHT FAIL: EKF HIGH IMU GYRO BIAS");
+			mavlink_log_critical(mavlink_log_pub, "Preflight Fail: EKF High IMU Gyro Bias");
 		}
 
 		success = false;
 		goto out;
 	}
 
 	// If GPS aiding is required, declare fault condition if the required GPS quality checks are failing
-	if (enforce_gps_required) {
-		bool ekf_gps_fusion = status.control_mode_flags & (1 << 2);
-		bool ekf_gps_check_fail = status.gps_check_fail_flags > 0;
-
-		if (!ekf_gps_fusion) {
-			// The EKF is not using GPS
-			if (ekf_gps_check_fail) {
-				// Poor GPS quality is the likely cause
-				if (report_fail) {
-					mavlink_log_critical(mavlink_log_pub, "PREFLIGHT FAIL: GPS QUALITY POOR");
-				}
-
-				gps_success = false;
-
-			} else {
-				// Likely cause unknown
-				if (report_fail) {
-					mavlink_log_critical(mavlink_log_pub, "PREFLIGHT FAIL: EKF NOT USING GPS");
-				}
+	if (enforce_gps_required || report_fail) {
+		const bool ekf_gps_fusion = status.control_mode_flags & (1 << estimator_status_s::CS_GPS);
+		const bool ekf_gps_check_fail = status.gps_check_fail_flags > 0;
 
-				gps_success = false;
-				gps_present = false;
-			}
+		gps_success = ekf_gps_fusion; // default to success if gps data is fused
 
-			success = false;
-			goto out;
+		const char *fail_str = enforce_gps_required ? "Fail" : "Warn";
```

Comment:
Let's use the existing mavlink severity levels instead.
https://mavlink.io/en/messages/common.html#MAV_SEVERITY

Reply:
https://mavlink.io/en/messages/common.html#STATUSTEXT